org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/launch/IMavenLauncher.java

```diff
@@ -22,6 +22,7 @@
  */
 public interface IMavenLauncher {
 
-  CompletableFuture<?> runMaven(File basedir, String goals, Properties properties) throws CoreException;
+  CompletableFuture<?> runMaven(File basedir, String goals, Properties properties, boolean interactive)
```

Comment:
Instead of a boolean, there could be 2 distinct methods `runBatch` and `runInteractive`. This would improve readability a bit (from "Clean Code" book).